mgmtfn/dockplugin/netDriver.go

```diff
@@ -523,3 +630,171 @@ func GetDockerNetworkName(nwID string) (string, string, string, error) {
 
 	return tenantName, netName, serviceName, nil
 }
+
+// FindGroupFromTag finds the group that has matching tag
+func FindGroupFromTag(epgTag string) (*mastercfg.EndpointGroupState, error) {
+	stateDriver, err := utils.GetStateDriver()
+	if err != nil {
+		return nil, err
+	}
+	epgCfg := &mastercfg.EndpointGroupState{}
+	epgCfg.StateDriver = stateDriver
+	epgList, err := epgCfg.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	var epg *mastercfg.EndpointGroupState
+	found := false
+	for _, epgP := range epgList {
+		epg = epgP.(*mastercfg.EndpointGroupState)
+		if epg.GroupTag == epgTag {
+			found = true
+			break
+		}
+	}
+	if !found {
+		err = errors.New("Couldn't find group matching the tag")
+		return nil, err
+	}
+	return epg, nil
+}
+
+// FindNetworkFromTag finds the network that has matching tag
+func FindNetworkFromTag(nwTag string) (*mastercfg.CfgNetworkState, error) {
+	stateDriver, err := utils.GetStateDriver()
+	if err != nil {
+		return nil, err
+	}
+	nwCfg := &mastercfg.CfgNetworkState{}
+	nwCfg.StateDriver = stateDriver
+	nwList, err := nwCfg.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	var nw *mastercfg.CfgNetworkState
+	found := false
+	for _, nwP := range nwList {
+		nw = nwP.(*mastercfg.CfgNetworkState)
+		if nw.NetworkTag == nwTag {
+			found = true
+			break
+		}
+	}
+	if !found {
+		err = errors.New("Couldn't find network matching the tag")
+		return nil, err
+	}
+	return nw, nil
+}
+
+// createNetworkHelper creates the association between docker network and contiv network
+//  if tag is given map docker net to epg or network, else create a contiv network
+func createNetworkHelper(networkID string, tag string, IPv4Data, IPv6Data []driverapi.IPAMData) error {
+	var tenantName, networkName, serviceName string
+	var err error
+	if tag != "" {
```

Comment:
tag is only for the docker network. ipam is tied to the network.
Docker supports only dual-stack IPv6. It expects IPv4 address as well.